projects/shared/src/lib/service/permissions.service.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2022 RERO
+ * Copyright (C) 2022 UCLouvain
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+import { PERMISSIONS } from '../util/permissions';
+
+@Injectable({
+  providedIn: 'any'
+})
+export class PermissionsService {
+
+  /** Permissions set */
+  private permissions = new Set<string>();
+
+  /**
+   * Set all permissions in set
+   * @param permissions - array of permissions
+   * @returns this
+   */
+  setPermissions(permissions: string[]): this {
+    permissions.map((permission: string) => {
+      if (!this.permissions.has(permission)) {
+        this.permissions.add(permission);
+      }
+    });
+
+    return this;
+  }
+
+  /**
+   * Can access
+   * @param permission - permission or array of permissions
+   * @param and - operator
+   * @returns true if permission is granted
+   */
+   canAccess(permission: string | string[], and: boolean = false): boolean {
```

Comment:
It's here the core of MY "andOperator" problem. If you change `and: boolean` by `operator: string = 'or'`, you could much more flexible if necessary in next PR's/US.